java-checks/src/main/java/org/sonar/java/checks/NoTestInTestClassCheck.java

```diff
@@ -163,6 +163,13 @@ private boolean isTestMethod(Symbol method) {
     Stream<Symbol.MethodSymbol> defaultMethodsFromInterfaces = symbol.interfaces().stream()
       .flatMap(i -> getAllMembers(i.symbol()))
       .filter(m -> ((JavaSymbol.MethodJavaSymbol) m).isDefault());
+    for (Symbol s : symbol.memberSymbols()) {
+      if (s instanceof Symbol.TypeSymbol) {
```

Comment:
I would prefer if you could use `s.isTypeSymbol()` instead. It would allow to remove the `instanceof` and stay with simpler usage of the semantic API for semantic checking.

You can also combine the two `if` statements in a single one with the conditions combined with `&&`.